drizztsaga/tpa/ds-core.tpa

```diff
@@ -9,62 +9,63 @@ PRINT @358
 
-  ACTION_IF GAME_IS ~BGT Tutu_TotSC~ BEGIN
+  ACTION_IF ((IDS_OF_SYMBOL (~animate~ ~GIANT_FROST~)) > 0) BEGIN
+     OUTER_SET "ANIM_FG" = IDS_OF_SYMBOL (~animate~ ~GIANT_FROST~)
+  END ELSE ACTION_IF ((IDS_OF_SYMBOL (~animate~ ~FROST_GIANT~)) > 0) BEGIN
+     OUTER_SET "ANIM_FG" = IDS_OF_SYMBOL (~animate~ ~FROST_GIANT~)
+  END ELSE BEGIN
+    COPY ~DRIZZTSAGA/Anim/MNO2~ ~override~      //FROST GIANT
     COPY_EXISTING ~ORC06.cre~      ~override~
               ~HLOLAF.cre~         ~override~
               ~CHAK.cre~           ~override~
               ~AR18FIG.cre~        ~override~
-        WRITE_LONG 0x28  0xE700   //IC_OROG1 instead of IC_OROG2
+      WRITE_LONG 0x28  0xE700   				//IC_OROG1 instead of IC_OROG2
     BUT_ONLY
   END
 
-  ACTION_IF ((GAME_IS ~BGT Tutu_TotSC~) AND ((IDS_OF_SYMBOL (~animate~ ~GIANT_FROST~)) < 0)) BEGIN
-    COPY ~DRIZZTSAGA/Anim/MNO2~ ~override~         //FROST GIANT
-  END ELSE 
-    OUTER_SET ANIME_FG = IDS_OF_SYMBOL (~animate~ ~GIANT_FROST~)
-
-  ACTION_IF ((GAME_IS ~BGT Tutu_TotSC~) AND ((IDS_OF_SYMBOL (~animate~ ~Cyclops~)) < 0)) BEGIN
-    COPY ~DRIZZTSAGA/Anim/MCYC~ ~override~         //Cyclop
+  ACTION_IF ((GAME_IS ~BGT Tutu_TotSC~) AND ((IDS_OF_SYMBOL (~animate~ ~CYCLOPS~)) < 0)) BEGIN
+    COPY ~DRIZZTSAGA/Anim/MCYC~ ~override~      //CYCLOP
   END
 
-  ACTION_IF ((GAME_IS ~BGT Tutu_TotSC~) AND ((IDS_OF_SYMBOL (~animate~ ~Marilith~)) < 0)) BEGIN
-    COPY ~DRIZZTSAGA/Anim/MTAN~ ~override~         //Merilith
+  ACTION_IF ((GAME_IS ~BGT Tutu_TotSC~) AND ((IDS_OF_SYMBOL (~animate~ ~MARILITH~)) < 0)) BEGIN
+    COPY ~DRIZZTSAGA/Anim/MTAN~ ~override~      //MARILITH
   END
-END
 
-ACTION_IF NOT (FILE_EXISTS ~data/CTB-RULE.bif~ OR FILE_EXISTS ~data/ROT-RULE.BIF~) THEN BEGIN
-  ACTION_IF ((GAME_IS ~BGT Tutu_TotSC~) AND ((IDS_OF_SYMBOL (~animate~ ~Harpy~)) < 0)) BEGIN
-    COPY ~DRIZZTSAGA/Anim/MTIC~ ~override~           //Harpy
+  ACTION_IF (IDS_OF_SYMBOL (~animate~ ~HARPY~) > 0) BEGIN
+    OUTER_SET "ANIM_HA" = IDS_OF_SYMBOL (~animate~ ~HARPY~)
   END ELSE 
-    OUTER_SET ANIME_HA = IDS_OF_SYMBOL (~animate~ ~Harpy~)
+    COPY ~DRIZZTSAGA/Anim/MTIC~ ~override~      //HARPY
 END
 
 ACTION_IF NOT (FILE_EXISTS ~data/ROT-RULE.BIF~) THEN BEGIN
-  ACTION_IF ((GAME_IS ~BGT Tutu_TotSC~) AND ((IDS_OF_SYMBOL (~animate~ ~Belhifet~)) < 0)) BEGIN
-    COPY ~DRIZZTSAGA/Anim/MUMB~ ~override~         //Belhifet
-  END ELSE
-    OUTER_SET ANIME_BE = IDS_OF_SYMBOL (~animate~ ~Belhifet~)
+  ACTION_IF (IDS_OF_SYMBOL (~animate~ ~BELHIFET~) > 0) BEGIN
+    OUTER_SET "ANIM_BE" = IDS_OF_SYMBOL (~animate~ ~BELHIFET~)
+  END ELSE 
+    COPY ~DRIZZTSAGA/Anim/MUMB~ ~override~      //BELHIFET
 END
 
-ACTION_IF GAME_IS ~BGEE~ BEGIN
+ACTION_IF GAME_IS ~BGEE~ AND (IDS_OF_SYMBOL (~animate~ ~CORNUGON~) > 0) BEGIN
+  OUTER_SET DECODE_SOUNDS = 1
   COPY ~DRIZZTSAGA/Anim/MCOR~ ~override~		// BGEE is missing Cornugon animations...
-  APPEND ~ANIMATE.IDS~ ~0xE0E0 IC_CORNUGONSKI~
+  APPEND ~ANIMATE.IDS~ ~0xE0E0 IC_CORNUGONSKI~  // CORNUGON
```

Comment:
This is a bad idea on the EEs. Yes, BGEE lacks the cornugon bams, but it has the animate.ids entry alongside the INI file, so it doesn't need to touch this.

Reply:
Exactly.